### ISSUE TITLE
When using Notepad++ as notepad replacement, it doesn't open a file without extension

### ISSUE THREAD

**Issue by kapsiR** [notepad-plus-plus/notepad-plus-plus#7902]
### Description of the Issue
When using Notepad++ as replacement for `notepad.exe`, the behavior for opening files without extension is weird.

### Steps to Reproduce the Issue
Using Notepad++ v7.8.4 x64
1. Replace default `notepad.exe` with Notepad++ (See [Notepad Replacement](https://npp-user-manual.org/docs/other-resources/))
2. Create a file **without** an extension, e.g. README
3. Try to open that file with `notepad.exe`
4. Error (it seems that Notepad++ tries to add `.txt` if ther is no file extension?!)

### Expected Behavior
Open files without extension too when using `notepad.exe` (e.g. README, LICENSE)

#### Proposed change
Add an additional check if the file without a file extension does exist or not before adding a `.txt` extension.
The [affected lines of code](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/8c6dc79df31fa4d97fab5fe7b07e366fb55f3dff/PowerEditor/src/winmain.cpp#L163-L166).
>		if ( *PathFindExtension(str.c_str()) == '\0' )
>		{
>			str.append(TEXT(".txt")); // If joined path has no extension, Notepad adds a .txt extension
>		}

### Actual Behavior
Notepad++ looks, if the same file with the `.txt` extension is available
1. if yes - it opens the wrong file (e.g. README.txt **instead** of README)
2. if no - it asks to create the file with a `.txt` extension

![grafik](https://user-images.githubusercontent.com/7165033/73427091-4f8a9a80-4336-11ea-9730-960f3003e3f0.png)

![grafik](https://user-images.githubusercontent.com/7165033/73426891-ca9f8100-4335-11ea-9c25-d7df0258d17e.png)

*Note: If I open any file without extension directly with Notepad++, it works as expected!*

### Debug Information
Notepad++ v7.8.4   (64-bit)
Build time : Jan 29 2020 - 01:31:03
Path : C:\Program Files\Notepad++\notepad++.exe
Admin mode : OFF
Local Conf mode : OFF
OS Name : Windows 10 Enterprise (64-bit) 
OS Version : 1909
OS Build : 18363.592
Plugins : DSpellCheck.dll mimeTools.dll NppConverter.dll NppExport.dll

**Comment by garfieldbanks** [comment]
Or is it that no one has idea what causes it to try to append a .txt at the end?

**Comment by garfieldbanks** [comment]
Or is it that no one has any idea what causes it to try to append a .txt at the end?

**Comment by kapsiR** [comment]
[This is the line of code](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/8c6dc79df31fa4d97fab5fe7b07e366fb55f3dff/PowerEditor/src/winmain.cpp#L165), that causes the problem:  

> str.append(TEXT(".txt")); // If joined path has no extension, Notepad adds a .txt extension

**Comment by garfieldbanks** [comment]
So we can just remove that line and commit it and then make a pull request to the main repository? Are you going to make a pull request or should I?

**Comment by garfieldbanks** [comment]
Would they accept it? Why does that line even exist?

**Comment by kapsiR** [comment]
When you look at the comments of the method, the question is, why is Notepad++ mirroring it's behavior and is it necessary?

> // 1. Converts /p to -quickPrint if it exists as the first parameter
> // 2. Concatenates all remaining parameters to form a file path, adding appending .txt extension if necessary
> // This seems to mirror Notepad's behaviour

Maybe @donho can help out here?

**Comment by mere-human** [comment]
> [This is the line of code](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/8c6dc79df31fa4d97fab5fe7b07e366fb55f3dff/PowerEditor/src/winmain.cpp#L165), that causes the problem:

This has been there from the time the function was created. See this commit:
https://github.com/notepad-plus-plus/notepad-plus-plus/commit/0db302953bb5a5ba3e06591fb721c8474ba9171e
So I think it makes sense to change that.

**Comment by mere-human** [comment]
By the way, is this case affected by "Hide extensions..." Explorer option?
![image](https://user-images.githubusercontent.com/9664141/110847763-fbcade80-82b5-11eb-9700-cae12203e014.png)

**Comment by garfieldbanks** [comment]
> By the way, is this case affected by "Hide extensions..." Explorer option?

No it is not.

**Comment by kapsiR** [comment]
I checked the default behavior of Notepad:  
![grafik](https://user-images.githubusercontent.com/7165033/110848707-ae973e80-82ae-11eb-89c3-fd1e18c52ef9.png)

It seems, the "only" issue here is, that it isn't checked if the file really exists.  
The behavior of Notepad is intended, but Notepad checks if the file exists and opens it:  
![grafik](https://user-images.githubusercontent.com/7165033/110848983-fcac4200-82ae-11eb-90dc-7ec782f349d8.png)

**Comment by kapsiR** [comment]
I just updated the issue with a *Proposed change* section 😉

**Comment by donho** [comment]
Has anyone tested the [solution of Notepad replacement](https://npp-user-manual.org/docs/other-resources/) under Windows 11 with v8.3.3 & v8.4? 
It doesn't work for me.

**Comment by kapsiR** [comment]
@donho I'll try to set up a quick test environment in the evening...

**Comment by kapsiR** [comment]
@donho It does not work by default on Windows 11, there is an additional DWORD that has to be set (changed):
```reg
"UseFilter"=dword:00000000
```

With this change, it works as before with Windows 10

**Comment by kapsiR** [comment]
@donho [As you can see](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7902#issuecomment-1116503895) it is not that easy for Windows 11. (the easiest solution would be to delete the Windows Notepad Store App)

Nevertheless it would be cool to get [the fix ](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11560)merged for all other Windows versions... 🙏

**Comment by rdipardo** [comment]
From this [related thread](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10653):

> See [zufuliu/notepad2#347 (comment)](https://github.com/zufuliu/notepad2/issues/347#issuecomment-913093135)
> 
> > As reported in issue [zufuliu/notepad2#367](https://github.com/zufuliu/notepad2/issues/367), following changes are required on Windows 11 (no need to uninstall UWP-based notepad).
> > ```
> > HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe
> >     Debugger                                REG_SZ      "C:\\Program Files\\Notepad++\\notepad++.exe -notepadStyleCmdline -z"
> >     UseFilter                               REG_DWORD   0
> >     0
> >         AppExecutionAliasRedirect           REG_DWORD   1
> >         AppExecutionAliasRedirectPackages   REG_SZ      *
> >         FilterFullPath                      REG_SZ      "notepad++.exe"
> >     1
> >         AppExecutionAliasRedirect           REG_DWORD   1
> >         AppExecutionAliasRedirectPackages   REG_SZ      *
> >         FilterFullPath                      REG_SZ      "notepad++.exe"
> >     2
> >         AppExecutionAliasRedirect           REG_DWORD   1
> >         AppExecutionAliasRedirectPackages   REG_SZ      *
> >         FilterFullPath                      REG_SZ      "notepad++.exe"
> > ```

**Comment by rdipardo** [comment]
[Also](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/10653#issuecomment-973176852):

> Hey guys, below you'll find a version of the .reg file for Windows 11 with **Notepad++ 32-bits**:
> 
> ```
> Windows Registry Editor Version 5.00
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe]
> "Debugger"="c:\\Program Files (x86)\\Notepad++\\notepad++.exe -notepadStyleCmdline -z"
> "UseFilter"="0"
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\0]
> "AppExecutionAliasRedirect"=dword:00000001
> "AppExecutionAliasRedirectPackages"="*"
> "FilterFullPath"="notepad++.exe"
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\1]
> "AppExecutionAliasRedirect"=dword:00000001
> "AppExecutionAliasRedirectPackages"="*"
> "FilterFullPath"="notepad++.exe"
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\2]
> "AppExecutionAliasRedirect"=dword:00000001
> "AppExecutionAliasRedirectPackages"="*"
> "FilterFullPath"="notepad++.exe"
> ```

**Comment by kapsiR** [comment]
@rdipardo I tried this on Win11 (OS Build 22000.613) and sadly it didn't work

**Comment by donho** [comment]
I wonder if  we should keep supporting Notepad replacement solution, since it becomes more complex  under Windows11 (and surly the future Windows' versions).

@pryrt What do you think?

**Comment by kapsiR** [comment]
In my opinion, it should still be an option, because:

- of all the Windows 10 users out there
- there is an relatively easy solution for Windows 11 users if they uninstall the store app

**Comment by pryrt** [comment]
@donho,

I tend to agree with @kapsiR : given how long the feature has been available, I am sure that users who have grown accustomed to that implementation would be upset if you took away that long-standing feature.

I would prefer to just add documentation to the [user manual](https://npp-user-manual.org/docs/other-resources/#notepad-replacement) which shows the [win11 workarounds](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/7902#issuecomment-1116503895) that @kapsiR had highlighted.

I used to wonder why people would use the hack, rather than File Associations or the more modern Open With mechanism. 
 But I learned a year or two ago that there are apps out there that call `notepad.exe Path\To\Filename.txt` rather than launching the default app for that extension, thus bypassing file-associations and OpenWith selections -- they use notepad.exe directly as an "external editor" to display log files, or do editing tasks that don't work well in their native GUI... and in those situations, being able to use the hack to replace MS notepad makes sense.

**Comment by kapsiR** [comment]
@donho Sorry for bothering (again) but did you made any decision regarding [this PR](https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11560)?

**Comment by donho** [comment]
@kapsiR 
> 3\. Try to open that file with `notepad.exe`

Sorry for the late reply.
I'm always trying reproduce this bug  in order to validate your PR.
By "Try to open that file with `notepad.exe`" could you provide the detail instructions for this step?

**Comment by donho** [comment]
After a closer look, I do believe this bug has been fixed in https://github.com/notepad-plus-plus/notepad-plus-plus/commit/6c3659fe7300992ae7a3cd32410377bd88ac263a

**Comment by kapsiR** [comment]
> After a closer look, I do believe this bug has been fixed in [6c3659f](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/6c3659fe7300992ae7a3cd32410377bd88ac263a)

@donho Thanks for taking your time to have another look here, I tested the behavior with the latest 0d22c4d867cfffb5da6a0567421fef2049f865ea and it works indeed! 🥳🙏🏼

**Comment by fubaWoW** [comment]
hi there,
i tried to open a File **without Extension** and it results in notepad++ try to append ".txt" to the file, but why?

i use this Registry Settings:
```ini
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe]
"Debugger"="\"C:\\Program Files\\Notepad++\\notepad++.exe\" -notepadStyleCmdline -z"
"UseFilter"=dword:00000000

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\0]
"AppExecutionAliasRedirect"=dword:00000001
"AppExecutionAliasRedirectPackages"="*"
"FilterFullPath"="notepad++.exe"

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\1]
"AppExecutionAliasRedirect"=dword:00000001
"AppExecutionAliasRedirectPackages"="*"
"FilterFullPath"="notepad++.exe"

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Image File Execution Options\notepad.exe\2]
"AppExecutionAliasRedirect"=dword:00000001
"AppExecutionAliasRedirectPackages"="*"
"FilterFullPath"="notepad++.exe"
```
Also tried without the Sub-Keys 0, 1 and 2 -> no difference!

Example, Run (Win + R) this:
```
"%SystemRoot%\system32\NOTEPAD.EXE" C:\Program Files (x86)\RivaTuner Statistics Server\Profiles\Global
```
will result in this:
```
---------------------------
Create new file
---------------------------
"C:\Program Files (x86)\RivaTuner Statistics Server\Profiles\Global.txt" doesn't exist. Create it?
---------------------------
Yes   No   
---------------------------
```

also when open a "*.reg" file by "Edit" in the Windows Context menu will result in this:
```
---------------------------
ERROR
---------------------------
Cannot open file " ".
---------------------------
OK   
---------------------------
```
BUT the file will still open?

Open "*.bat", "*.ini" or "*.xml" files with "Edit" from the Windows Context menu works without errors on the other hand?!

any solution?
